SubMotionVR Mod/SubMotion/SubMotion/Patchers/ArmsController.cs

```diff
@@ -152,103 +88,48 @@ public static void Postfix(ArmsController __instance)
             if ((Player.main.motorMode != Player.MotorMode.Vehicle && !player.cinematicModeActive) || pda.isActiveAndEnabled)
             {
-                VRHandsController.instance.UpdateHandPositions(ik);
+                VRHandsController.main.UpdateHandPositions();
             }
         }
     }
 
     [HarmonyPatch(typeof(ArmsController))]
     [HarmonyPatch("Reconfigure")]
-    internal class ArmsController_Reconfigure_Patch
+    class ArmsController_Reconfigure_Patch
     {
-        private static Player _player;
-        public static Player player
-        {
-            get
-            {
-                if (_player == null)
-                {
-                    _player = global::Utils.GetLocalPlayerComp();
-                }
-                return _player;
-            }
-        }
-        private static PDA ___pda;
-        public static PDA pda
-        {
-            get
-            {
-                if (___pda == null)
-                {
-                    ___pda = player.GetPDA();
-                }
-                return ___pda;
-            }
-        }
-        private static FullBodyBipedIK ___ik;
-        public static FullBodyBipedIK ik
-        {
-            get
-            {
-                if (___ik == null)
-                {
-                    ___ik = ik.GetComponent<FullBodyBipedIK>();
-                }
-                return ___ik;
-            }
-        }
-
-        private static Transform leftWorldTarget;
-        public static Transform leftTarget
-        {
-            get
-            {
-                if (leftWorldTarget == null)
-                {
-                    leftWorldTarget = leftTarget;
-                }
-                return leftWorldTarget;
-            }
-        }
-
-        private static Transform rightWorldTarget;
-        public static Transform rightTarget
-        {
-            get
-            {
-                if (rightWorldTarget == null)
-                {
-                    rightWorldTarget = rightTarget;
-                }
-                return rightWorldTarget;
-            }
-        }
-
         [HarmonyPrefix]
         public static void Prefix(ArmsController __instance, PlayerTool tool)
         {
-            ___ik.solver.GetBendConstraint(FullBodyBipedChain.LeftArm).bendGoal = __instance.leftHandElbow;
-            ___ik.solver.GetBendConstraint(FullBodyBipedChain.LeftArm).weight = 1f;
+            FullBodyBipedIK ik = VRHandsController.main.ik;
+
+            ik.solver.GetBendConstraint(FullBodyBipedChain.LeftArm).bendGoal = __instance.leftHandElbow;
+            ik.solver.GetBendConstraint(FullBodyBipedChain.LeftArm).weight = 1f;
             if (tool == null)
             {
-                Traverse.Create(__instance).Field("leftAim").Field("shouldAim").SetValue(false);
-                Traverse.Create(__instance).Field("rightAim").Field("shouldAim").SetValue(false);
+                Traverse tInstance = Traverse.Create(__instance);

```

Comment:
this is the magic that lets us access private fields, otherwise C# won't let us touch them